include/openthread/nat64.h

```diff
@@ -390,6 +390,33 @@ void otNat64SetReceiveIp4Callback(otInstance *aInstance, otNat64ReceiveIp4Callba
  */
 otError otNat64GetCidr(otInstance *aInstance, otIp4Cidr *aCidr);
 
+/**
+ * This function pointer is called when the state of a component of NAT64 changed.
+ *
+ * @param[in]  aContext  A pointer to application-specific context.
+ * @param[in]  aNewState The new state of the component
+ *
+ */
+typedef void (*otNat64StateChangeCallback)(void *aContext, otNat64State aNewState);
+
+/**
+ * Registers a callback receiving NAT64 translator state change event.
+ *
+ * Note: This function is meant to be used by platforms to perform platform specific operations (like setting up
+ * routes).
+ *
+ * Available when `OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE` is enabled.
+ *
+ * @param[in]  aInstance         A pointer to an OpenThread instance.
+ * @param[in]  aCallback         A pointer to a function that is called when the state of NAT64 translator
+ *                               is changed.
+ * @param[in]  aCallbackContext  A pointer to application-specific context.
+ *
+ */
+void otNat64SetTranslatorStateChangeCallback(otInstance                *aInstance,
```

Comment:
Should we use the existing state changed API? https://github.com/openthread/openthread/blob/b02bdb9e3bfc0207737bebb94824713f61f5352c/include/openthread/instance.h#L229

I know we are short on 32-bit flags, but we should address that separately.